src/rc_ptr.c

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+
+#include "c_pal/thandle.h"
+
+#include "c_util/rc_ptr.h"
+
+THANDLE_TYPE_DEFINE(RC_PTR);
+
+static void rc_ptr_destroy(RC_PTR* rc_ptr)
+{
+    if (rc_ptr == NULL)
+    {
+        LogError("Invalid arguments: RC_PTR* rc_ptr=%p", rc_ptr);
+    }
+    else
+    {
+        rc_ptr->free_func((void*)rc_ptr->ptr);
+    }
+}
+
+THANDLE(RC_PTR) rc_ptr_create_with_move_memory(const void* ptr, RC_PTR_FREE_FUNC free_func)
```

Comment:
>free_func [](http://example.com/codeflow?start=81&length=9)

Frankly, free_func might be `NULL` and still be valid, as a courtesy to the user.

What is the user passes some memory here that's stack allocated and that doesn't have a "free" function?